daemon/container.go

```diff
@@ -1052,7 +1052,7 @@ func (container *Container) setupContainerDns() error {
 		return err
 	}
 
-	if config.NetworkMode != "host" {
+	if !config.NetworkMode.IsHost() || !config.NetworkMode.IsContainer() {
```

Comment:
Also I think this would need to be `&&`, not `||`, since if anytime it's not host it wouldn't run the 2nd check anyway.